Makefile

```diff
@@ -36,9 +37,13 @@ misc: tools
 	mkdir -p $(MISC_OUT)
 	make -C $(T)/misc OUT_DIR=$(MISC_OUT)
 
+doc:
```

Comment:
> What's the reason to change where the documentation build output goes?

So that it's consistent with the rest of the build, all build artefacts are put under the build dir. (except I could get `doxygen` to put its stuff in there so it's still done inside the `doc` folder.

Reply:
> The doc/Makefile won't notice your change here because it sets:
> `BUILDDIR      = _build`
> To complete this change, you'd need to tweak the doc/Makefile to use
> `BUILDDIR      ?= _build`

That was not needed on my system. With this patch, calling `make doc` from the top-level folder puts the final documentation inside `build`. Using `O=xxx` works too. Have you tried this on your side?